pkg/knuu/instance_helper.go

```diff
@@ -274,8 +243,81 @@ func (i *Instance) destroyFiles() error {
 	return nil
 }
 
+// deployResources deploys the resources for the instance
+func (i *Instance) deployResources() error {
+	if len(i.portsTCP) != 0 || len(i.portsUDP) != 0 {
+		logrus.Debugf("Ports not empty, deploying service for instance '%s'", i.k8sName)
+		svc, _ := k8s.GetService(k8s.Namespace(), i.k8sName)
+		if svc == nil {
+			err := i.deployService()
+			if err != nil {
+				return fmt.Errorf("error deploying service for instance '%s': %w", i.k8sName, err)
+			}
+		} else if svc != nil {
+			err := i.patchService()
+			if err != nil {
+				return fmt.Errorf("error patching service for instance '%s': %w", i.k8sName, err)
+			}
+		}
+	}
```

Comment:
this should be refactored into separate private functions

Reply:
https://github.com/celestiaorg/knuu/pull/168/commits/0326589d701f1321c388d9056049a6a621fc9017